src/AnalyzeView/MAVLinkMessage.cc

```diff
@@ -108,7 +108,7 @@ void QGCMAVLinkMessage::update(const mavlink_message_t &message)
     _count++;
     _message = message;
 
-    if (_selected) {
+    if (_selected || _fieldSelected) {
```

Comment:
Hmm, interesting. Shouldn't this just be _fieldSelected? Since the message might be selected but that doesn't mean that a field is selected. Otherwise you end up doing a bunch of work in updateField every time you select a new top level message. Or am I reading the code wrong?

Reply:
You're right, `_selected` is not needed. It's not a big overhead though because there's only one message selected at the time. 

Do you want me to remove `_selected` from the if statement?

Reply:
Yes, you should remove it. Make the code clearer to others looking at it later.

Reply:
Any updates on this?